common/persistence/persistence-tests/historyV2PersistenceTest.go

```diff
@@ -260,11 +260,21 @@ func (s *HistoryV2PersistenceSuite) TestReadBranchByPagination() {
 	req.NextPageToken = resp.NextPageToken
 
```

Comment:
same here, we need to add a test to assert on EntityNotExistsError so that we can make sure this behavior won't be broken

Reply:
Done.